resources/js/shared/components/cities/CitiesTable.tsx

```diff
@@ -0,0 +1,226 @@
+import { useEffect, useState } from "react";
+
+import { deleteCity, getCities } from "~/api/cities";
+import { CreateCityModal } from "~/modals/City/CreateCityModal";
+import { EditCityModal } from "~/modals/City/EditCityModal";
+import { Button } from "~/ui";
+
+interface Pagination {
+  count: number;
+  total: number;
+  perPage: number;
+  currentPage: number;
+  totalPages: number;
+  links: {
+    next: string;
+    previous: string;
+  };
+}
+interface Filters {
+  [key: string]: string;
+}
+
+export const CitiesTable = () => {
+  const [loading, setLoading] = useState<boolean>(true);
+  const [cities, setCities] = useState<any[]>([]);
+  const [pagination, setPagination] = useState<Pagination | null>(null);
+  const [sort, setSort] = useState<string | null>(null);
+  const [order, setOrder] = useState<string>("asc");
+  const [filters, setFilters] = useState<Filters>({});
+  const [isEditModalVisible, setIsEditModalVisible] = useState<boolean>(false);
+  const [isCreateModalVisible, setIsCreateModalVisible] =
+    useState<boolean>(false);
+  const [selectedCity, setSelectedCity] = useState<any | null>(null);
+
+  const buildQueryParams = () => {
+    const params = new URLSearchParams();
+
+    Object.keys(filters).forEach((key) => {
+      params.append(`filter[${key}]`, filters[key]);
+    });
+
+    if (sort) params.append("sort", order === "asc" ? sort : `-${sort}`);
+
+    return params.toString();
+  };
+
+  const loadCities = async (url: string | null = null) => {
+    try {
+      const queryParams = buildQueryParams();
+      const res = url
+        ? await getCities(url, queryParams)
+        : await getCities(null, queryParams);
+      setCities(res.data);
+      setPagination(res.pagination);
+    } catch (error: any) {
+      console.error("Error obtaining cities:", error.message);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleSortChange = (e: any) => {
```

Comment:
```suggestion
  const handleSortChange = (e: ChangeEvent<HTMLInputElement>) => {
```

You can type this event like this